doc/Plugin-API.md

```diff
@@ -167,14 +176,45 @@ The `Ext` side then gets the cached object, executes appropriate functions and r
 Another example to browse are the [TaskExtImpl](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/plugin/tasks/tasks.ts) and [TaskMainImpl](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/tasks-main.ts) classes.
 
 To [ensure correct type conversion](#encoding-and-decoding-rpc-messages) between the Theia backend and the plugin host we define an API protocol based on types and DTOs that can be transmitted safely.
-The plugin API and it's types are defined in [plugin-ext/src/common/plugin-api-rpc.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-api-rpc.ts) with some additional conversion on the `Ext` side being defined in [plugin-ext/src/plugin/type-converters.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/plugin/type-converters.ts).
+The plugin API and its types are defined in [plugin-ext/src/common/plugin-api-rpc.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-api-rpc.ts) with some additional conversion on the `Ext` side being defined in [plugin-ext/src/plugin/type-converters.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/plugin/type-converters.ts).
 Thus, this is also a good starting point to look for conversion utilities for existing types.
 
 ### Adding new types
 
 New classes and other types such as enums are usually implemented in [plugin-ext/src/plugin/types-impl.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/plugin/types-impl.ts).
 They can be added there and then we can add them to the API object created in the API factory.
 
+## Headless Plugins
+
+The majority of plugin use cases are for extension of the Theia user experience via the VS Code compatible API provided by Theia.
+These plugins use either the `vscode` API object if they use the `"vscode"` engine type in their package manifests or else the `theia` object if they use the `"theiaPlugin"` engine.
+The lifecycle of these kinds of plugins is bound to _frontend connections_: for each connected frontend, the Theia backend spawns a plugin host host process dedicated to it in which these plugins are loaded and activated (as applicable to their declared activation events).
+In the plugin host for a frontend connection these plugins have access to Theia services as discussed above, isolated from Theia itself and from all of the other instances of the same plugin running in plugin hosts for all other frontend connections.
+
+Headless plugins, by contrast, are quite different in most respects:
+
+- they are encapsulated in a single plugin host process in the backend, separate from all frontend-connection plugin hosts.
```

Comment:
```suggestion
- They are encapsulated in a single plugin host process in the backend, separate from all frontend-connection plugin hosts.
```

Should we start with a capital letter if we have a complete sentence? I'd also make sure that each sentense then ends with a dot.

Reply:
There are different styles for interpretation of bullet lists as complete sentences or not. I'll try harder to follow the established pattern in these docs!